### PR TITLE
Render Errors to screen

### PR DESCRIPTION
**Pull request by andrewMacmurray**
+ Relevant errors are now shown to client
+ Fix bug where getClientActions was only returning one set of actions
+ functions that update actions and client actions UI now in callbacks (so they don't go out of sync with localStorage)

#118

### CODE REVIEW

File: lib/db/data.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 const db = (connection) => {
   const db = {};
   const query = (query) => (values = [], cb) => connection.query(query, values, cb);
+  const multiple = (query) => (values = [], cb) => connection.query(values.map(_ => query).join(''), values, cb);
+  const flatten = (arr) => arr.reduce((a, b) => a.concat(b), []);
```
Comment (rjmk) [inline lib/db/data.js]: Be aware with the slightly dodgy behaviour of this implementation of flatten -- it treats inner arrays and single elements as the same

`flatten([[1], [2], 3]) // [1, 2, 3]`

File: lib/db/data.js [hunk 2]
```diff
@@ -15,8 +17,8 @@ const db = (connection) => {
   db.getPrompts = query('SELECT * FROM prompts');
   db.getPromptByValue = query('SELECT * FROM prompts WHERE value = ?');
   db.getActions = query('SELECT * FROM actions WHERE locationid = ?');
-  db.getClientActions = query('SELECT * FROM client_actions WHERE clientid = ?');
-  db.deleteClientActions = query('DELETE FROM client_actions WHERE clientid = ?');
+  db.getClientActions = multiple('SELECT * FROM client_actions WHERE clientid = ?;');
```
Comment (rjmk) [inline lib/db/data.js]: Let's have semicolons after all the queries (it's simpler than having to remember that they're necessary for `multiple`)

File: lib/errorMessages.js [hunk 3]
```diff
@@ -2,10 +2,13 @@ const Boom = require('boom');
 
 module.exports = {
   login: {
-    dbError: Boom.create(500, 'database error during login, please contact the product owner'),
-    serverError: Boom.create(500, 'server error during login, please contact the product owner'),
-    getDataError: Boom.create(500, 'database error when getting data after login, please contact product owner'),
+    dbError: Boom.create(400, 'database error during login, please contact the product owner'),
```
Comment (rjmk) [inline lib/errorMessages.js]: Why the change in code? This seems wrong

File: public/src/js/helpers.js [hunk 4]
```diff
@@ -13,6 +13,20 @@ utils.firstLetterCaps = function (string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 };
 
+utils.removeItemAtIndex = function (list, i) {
+  return list.slice(0, i).concat(list.slice(i + 1));
```
Comment (rjmk) [inline public/src/js/helpers.js]: `list.filter(function (_, j) { return i === j })` might be a bit clearer

**Comment by rjmk** [comment]
This is looking pretty good. Can we resolve conflicts?

File: lib/errorMessages.js (hunk 3 above)
Comment (andrewMacmurray) [inline lib/errorMessages.js]: We changed this as Boom won't let you show 500 errors to the client other than `Internal server error`, will have another go with 500s and see if we can get something useful showing.

File: public/src/js/helpers.js (hunk 4 above)
Comment (andrewMacmurray) [inline public/src/js/helpers.js]: nice! is that supposed to be `i !== j` though?

File: lib/db/data.js (hunk 1 above)
Comment (andrewMacmurray) [inline lib/db/data.js]: I see what you mean. In this file we only ever want a single array coming out of a query, would it be safer to make it recursively merge arrays down to a single array?
